spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -315,6 +338,99 @@ contract RolloverVault is
         return redemptions;
     }
 
+    /// @notice Swaps the bond tranche slices from the user into the underlying collateral for a fee,
+    ///         given that the vault has the remaining slices for immature redemption.
+    /// @dev When a user has some tranche slices from a given bond, and the vault has the remaining slices,
+    ///      this method allows the user and the vault to "meld" their assets together to redeem the underlying collateral
+    ///      from the bond. The user's share of the underlying is returned back and the vault charges
+    ///      the user a fee for providing liquidity.
+    /// @param bond The bond whose tranches are to be swapped.
+    /// @param trancheAmtsIn A list of amounts of each bond tranche the user is to deposit.
+    ///                      When the user does not have a particular slice, its amount in the list is to be set to zero.
+    /// @return The amount of underlying tokens returned.
+    function meld(IBondController bond, uint256[] memory trancheAmtsIn)
+        external
+        nonReentrant
+        whenNotPaused
+        returns (uint256)
+    {
+        // get bond tranches
+        BondTranches memory bt = bond.getTranches();
+
+        // 1) We check that the bond's underlying collateral token matches the vault's underlying.
+        // 2) We validate that the bond has at least 2 tranches to perform the meld operation.
+        // 3) We validate that the bond hasn't reached maturity as matching only works through immature redemption.
+        // 4) We check if the given bond is NOT malicious, by examining if at least one of the bond's tranches
+        //    is already in the vault from previous deployments. And if the parent bond of the tranche already in the vault
+        //    is the given bond.
+        if (
+            bond.collateralToken() != address(underlying) ||
+            bt.tranches.length <= 1 ||
+            bond.isMature() ||
```

Comment:
I think it's possible for the bond to be past the date of maturity, but still `isMature() == false` because the mature function hasn't been called yet.

I think I'm ok with this for gas savings, it just means that we're still doing a "immature redeem" on a bond that theoretically could be "mature redeemed".

Reply:
Yes, Its possible for the bond to be past maturity but `isMature` to be false. In which case the `bond.redeem` (ie the immature redemption) will execute as intended and not revert. The current flow should work as is. 

Alternatively we could even get rid of this check and rely on the `bond.redeem` to revert if the bond `isMature` ..